### ISSUE TITLE
token can not custom

### ISSUE THREAD

**Issue by nickchu852** [progressivetech/net.ourpowerbase.qrcodecheckin#33]
I can use {qrcodecheckin.qrcode_html_5} to display a QR code, but if I want to change it based on different event IDs, {qrcodecheckin.qrcode_html_{event.id}} cannot be used normally. Please tell me how to solve it.

**Comment by jmcclelland** [comment]
I don't think you can nest tokens like that, so I don't think what you are trying to do is possible.

**Comment by nickchu852** [comment]
So this PLUGIN can only specify one activity, and cannot make changes because of different event.id?

**Comment by jmcclelland** [comment]
The full explanation for how it works is in the [readme file](https://github.com/progressivetech/net.ourpowerbase.qrcodecheckin). 

In short - this extension operates on events. In the event configuration screen, you can choose which events should have QR codes enabled. If an event has the Enable QR codes check box checked, then a token is automatically generated for you and will be available in the list of tokens.

So, rather than try to create your own custom tokens, just check the box for your event, and the token will appear in the drop down list.

**Comment by nickchu852** [comment]
If more than one event is enabled for QR code usage, it becomes impossible to use it as the email form, Events - Registration Confirmation and Receipt (on-line), is sent upon completing the event application.

For example, if "Event 1" and "Event 2" are both open for registration, the same email form will be used when sending the confirmation email. However, the tokens used for the QR code display are not flexible and can only be {qrcodecheckin.qrcode_html_1} or {qrcodecheckin.qrcode_html_2}. It is not possible to dynamically change the event number using the format {qrcodecheckin.qrcode_html_{event.id}}.

I have already spent a lot of time trying to understand and solve this issue, but I haven't been successful. Even in your GitHub usage examples, you use {qrcodecheckin.qrcode_html_3}, where the number 3 is hard-coded and not flexible. I hope you understand what I mean.

**Comment by MegaphoneJon** [comment]
Noting that a generic participant token now exists so this can be closed now.

**Comment by jmcclelland** [comment]
Great - thanks for the follow up.
